libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -57,6 +57,23 @@ bool DocStringAnalyser::visit(FunctionDefinition const& _function)
 	return true;
 }
 
+bool DocStringAnalyser::visit(VariableDeclaration const& _variable)
+{
+	if (_variable.isStateVariable())
+	{
+		static set<string> const validPublicTags = set<string>{"dev", "notice"};
+		if (_variable.isPublic())
+			parseDocStrings(_variable, _variable.annotation(), validPublicTags, "public state variables");
+		else
+		{
+			parseDocStrings(_variable, _variable.annotation(), validPublicTags, "non-public state variables");
+			if (_variable.annotation().docTags.find("notice") != _variable.annotation().docTags.end())
+				m_errorReporter.warning("Documentation tag @notice on non-public state variables will be disallowed in 0.7.0.");
```

Comment:
This error is also shown on docstrings that do not use `@notice` explicitly, right? I think we should change the text then.

Reply:
Another option would also to just change the default tag to `@dev` for non-public state variables.

Reply:
@aarlt was this handled?